app/modules/asset-reminder/service.server.ts

```diff
@@ -0,0 +1,330 @@
+    const { perPage } = cookie;
+
+    const skip = page > 1 ? (page - 1) * perPage : 0;
+    const take = perPage >= 1 && perPage <= 100 ? perPage : 20;
+
+    const finalWhere: Prisma.AssetReminderWhereInput = {
+      organizationId,
+      ...where,
+    };
+
+    if (search) {
+      const searchTerms = search
+        .toLowerCase()
+        .trim()
+        .split(",")
+        .map((term) => term.trim())
+        .filter(Boolean);
+
+      finalWhere.OR = searchTerms.map((term) => ({
+        OR: [
+          { name: { contains: term, mode: "insensitive" } },
+          { message: { contains: term, mode: "insensitive" } },
+          {
+            teamMembers: {
+              some: {
+                user: {
+                  OR: [
+                    {
+                      firstName: {
+                        contains: term,
+                        mode: "insensitive",
+                      },
+                    },
+                    {
+                      lastName: {
+                        contains: term,
+                        mode: "insensitive",
+                      },
+                    },
+                  ],
+                },
+              },
+            },
+          },
+          {
+            asset: { title: { contains: term, mode: "insensitive" } },
+          },
+        ],
+      }));
+    }
+
+    const [reminders, totalReminders] = await Promise.all([
+      db.assetReminder.findMany({
+        where: finalWhere,
+        take,
+        skip,
+        include: ASSET_REMINDER_INCLUDE_FIELDS,
+        orderBy: { [orderBy ?? "alertDateTime"]: orderDirection ?? "desc" },
+      }),
+      db.assetReminder.count({ where: finalWhere }),
+    ]);
+
+    const totalPages = Math.ceil(totalReminders / perPageParam);
+
+    return {
+      reminders,
+      totalReminders,
+      page,
+      perPage,
+      totalPages,
+      search,
+    };
+  } catch (cause) {
+    throw new ShelfError({
+      cause,
+      message: "Something went wrong while getting asset reminders.",
+      label,
+    });
+  }
+}
+
+export async function editAssetReminder({
+  id,
+  name,
+  message,
+  alertDateTime,
+  organizationId,
+  teamMembers,
+}: Pick<
+  AssetReminder,
+  "id" | "name" | "message" | "alertDateTime" | "organizationId"
+> & { teamMembers: TeamMember["id"][] }) {
+  try {
+    await validateTeamMembersForReminder(teamMembers);
+
+    /** This will act as a validation to check if reminder exists */
+    const reminder = await db.assetReminder.findFirstOrThrow({
+      where: { id, organizationId },
+    });
+
+    const now = new Date();
+    if (now > reminder.alertDateTime) {
+      throw new ShelfError({
+        cause: null,
+        message: "Edit is not allowed for this reminder.",
+        label: "Asset Reminder",
+        additionalData: { id },
+        shouldBeCaptured: false,
+      });
+    }
+
+    const updatedReminder = await db.assetReminder.update({
+      where: { id: reminder.id },
```

Comment:
missing `organizationId`

Reply:
I already have a validation to check if the reminder already exists or not with the `organizationId`
Here 
![image](https://github.com/user-attachments/assets/4b4c491b-b93c-4ada-a625-a896c2282d93)